### PR TITLE
A lot of small optimizations and code refactor

### PR DESCRIPTION
**Pull request by GermanAizek**
@IntelOrca, @Gymnasiast, @duncanspumpkin
check my PR please and give feedback.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Split this up into different pr's. Why do you think deques would be better some of those don't make much sense.

File: src/openrct2-ui/audio/AudioContext.cpp [hunk 1]
```diff
@@ -44,9 +44,9 @@ namespace OpenRCT2::Audio
             return _audioMixer;
         }
 
-        std::vector<std::string> GetOutputDevices() override
+        std::deque<std::string> GetOutputDevices() override
```
Comment (duncanspumpkin) [inline src/openrct2-ui/audio/AudioContext.cpp:47]: Stick to vector this. This is only used in like one, two places it's only ever about 2 elements big.

**Comment by GermanAizek** [comment]
> Split this up into different pr's. Why do you think deques would be better some of those don't make much sense.

Ok, I will recreate PRs

**Comment by Broxzier** [comment]
I presume you used some static analyzer like CppCheck to find these. Stick to one type of change per PR.

**Pull request closed without merging**